syntax = "proto3";

service Translator {
  rpc StreamTranslate (stream TranslationRequest) returns (stream TranslationResponse);
}

message TranslationRequest {
  string text = 1;
  string direction = 2; // KR_EN or EN_KR
}

message TranslationResponse {
  string translation = 1;
  double latency_ms = 2;
  int32 core_count = 3;
}